src/brain/util/tf_session_entity/tf_session_entity.cpp

```diff
@@ -119,13 +124,20 @@ OutputType *TFSE_TYPE::Eval(
                 nullptr, 0,                                     // Operations
                 nullptr, status_);
   PELOTON_ASSERT(TF_GetCode(status_) == TF_OK);
+  for (TF_Tensor *in_val : in_vals) {
+    TF_DeleteTensor(in_val);
+  }
+  for (TF_Tensor *out_val : out_vals) {
+    tensor_gc_.push_back(out_val);
+  }
```

Comment:
I actually don't think TfSession should manage the memory for input and output. It doesn't have those fields as member variables, nor does the session owns the input and output. The user who uses the session should manage the input and output by themselves. I think adding a GC mechanism here is complicating the programming logic.